src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -118,6 +239,48 @@
      */
     Status checkInput(FilterInfo filterInfo);
 
+    /**
+     * Returns a filter that merges the status of this filter and another filter.
+     * If the other filter is {@code null}, this filter is returned.
+     * Otherwise, a filter is returned to merge the pair of {@code non-null} filters.
+     * When used as an ObjectInputFilter by invoking the {@link ObjectInputFilter#checkInput} method,
+     * the result is:
+     * <ul>
+     *     <li>{@link Status#REJECTED}, if either filter returns {@link Status#REJECTED}, </li>
+     *     <li>Otherwise, {@link Status#ALLOWED}, if either filter returned {@link Status#ALLOWED}, </li>
+     *     <li>Otherwise, return {@link Status#UNDECIDED}</li>
+     * </ul>
+     *
+     * @param otherFilter a filter to be merged with this filter, may be null
+     * @return an {@link ObjectInputFilter} that merges the status of this and another filter
+     */
+    default ObjectInputFilter merge(ObjectInputFilter otherFilter) {
+        return (otherFilter == null) ? ObjectInputFilter.this : new Config.MergeFilter(this, otherFilter);
+    }
+
+    /**
+     * Returns a filter, that when applied to this filter that is checking a class, maps
+     * {@code Status.UNDECIDED} to {@code Status.REJECTED}, otherwise returns the status of the other filter.
+     * Object serialization accepts a class if the filter returns {@code UNDECIDED}.
+     * Appending a filter to reject undecided results for classes that have not been
```

Comment:
Could this be rephrased? I am not sure what "Appending" means in this context: more specifically appending to what? Maybe there's a way to  describe that in terms of "Mapping" rather than "Appending"?